### DISCUSSION TITLE
How to create an email using CLI tmail-backend:memory-branch-master ?

### DISCUSSION THREAD

**Discussion by aioun** [linagora/tmail-flutter#1396] [General]
Dear all,
I want to create an email directly into  the server using the CLI commands. I tested some commands in provisioning.sh bu I am looking for the command that allow me to create an email. I could be one of these: CopyMailbox <srcBean> <dstBean>, or ImportEml <namespace> <user> <name> <path>. 
Could you please give me a detailed example?
Thank You.

**Comment by chibenwa** [comment]
Hello @aioun 

ImportEml takes one of the EML file mime message on the file system and imports it into one mailbox. 

Have a look at https://james.apache.org/server/manage-cli.html for more details.

**Reply by aioun** [reply]
Thanks. However, this doesn't respond to my needs. I want to send a message from one user to another without using the frontend (interface)?
Is there a solution using CLI ?

**Reply by chibenwa** [reply]
No james cli do not allow that but surely tons of command line tools allows doing this. 

https://tecadmin.net/ways-to-send-email-from-linux-command-line/

**Reply by aioun** [reply]
Thanks. Here is the trace if any one is interested.
telnet my-jmap-server.com 3587
Trying 127.0.0.1...
Connected to my-jmap-server.com.
Escape character is '^]'.
220 Apache JAMES awesome SMTP Server
ehlo my-jamp.server.com
250-292c0e657840 Hello my-jamp.server.com [172.19.0.1])
250-AUTH LOGIN PLAIN
250-AUTH=LOGIN PLAIN
250-PIPELINING
250-ENHANCEDSTATUSCODES
250-8BITMIME
250 STARTTLS
mail from: <bob@my-jamp-server.com>           
250 2.1.0 Sender <bob@my-jamp-server.com> OK
rcpt to: <alice@my-jmap-server.com>   
250 2.1.5 Recipient <alice@my-jmap-server.com> OK
data
354 Ok Send data ending with <CRLF>.<CRLF>
hello how are you?
.
250 2.6.0 Message received
quit

**Comment by aioun** [comment]
The emails sent via telnet do not appear in the sender box. Besides, Is there a way to persist the emails in memory branch server of tmail backend? 
The idea behind it, is to be able to populate the server with a dateset automatically for further usage. 
Thanks,

**Comment by chibenwa** [comment]
> Besides, Is there a way to persist the emails in memory branch server of tmail backend?

Using the distributed server. https://github.com/linagora/tmail-backend/tree/master/tmail-backend/apps/distributed

> No james cli do not allow that but surely tons of command line tools allows doing this.

Well, I mis-understood your question. I thought you want to relay emails to third party servers viaa SMTP, but you just want to append emails with From header field ALice to Bob inbox where both are local users.

ImportEml from the CLI is sufficient for that goal: just position the right headers upon importing the EML would do the trick, as the SMTP transport envelope is not stored in user mailboxes.
